src/mudlet-lua/lua/geyser/GeyserAdjustableContainer.lua

```diff
@@ -0,0 +1,833 @@
+--Adjustable Container
+--Just use it like a normal Geyser Container with some extras like: 
+--moveable, adjustable size, attach to borders, minimizeable, save/load...
+--right click on top border for menu
+--Inspired heavily by Adjustable Label (by Jor'Mox ) and EMCO (by demonnic )
+--by Edru 2020
+
+Adjustable = Adjustable or {}
+
+--- Adjustable Container
+-- @class table
+-- @name Adjustable.Container
+Adjustable.Container = Adjustable.Container or Geyser.Container:new({name = "AdjustableContainerClass",})
+
+local adjustInfo = {}
+
+--- Internal function to add "%" to a value and round it
+-- @param num is the value a "%" will be added to
+local function make_percent(num)
+    num = math.floor(10000*num)/100
+    num = tostring(num).."%"
+    return num
+end
+
+--- Internal function: checks where the mouse is at on the Label 
+-- and saves the information for further use at resizing/repositioning
+-- also changes the mousecursor for easier use of the resizing/repositioning functionality
+-- @param self the Adjustable.Container it self
+-- @param label the Label which allows the Container to be adjustable
+-- @param event Mouse Click event and its infomations
+local function adjust_Info(self, label, event)
+    local x, y = getMousePosition()
+    local w, h = self.adjLabel:get_width(), self.adjLabel:get_height()
+    local x1, y1 = x - event.x, y - event.y
+    local x2, y2 = x1 + w, y1 + h
+    local left, right, top, bottom = event.x <= 10, x >= x2 - 10, event.y <= 3, y >= y2 - 10
+    if right and left then left = false end
+    if top and bottom then top = false end
+    
+    if event.button ~= "LeftButton" and not self.minimized then
+        if (top and not left and not right) or (bottom and not left and not right) then
+            label:setCursor("SizeVer")
```

Comment:
Maybe we can improve on the naming here - this doesn't read all that simple

Reply:
I'm not sure how the naming could be better but I cleaned up a bit the if statements.
It should be better readable now.

Reply:
Oh I meant the cursor. What size "size ver" mean? Not much without looking it up...

Reply:
That's not for this PR, I should have clarified. Just looking at how the new cursors look in real code :)

Reply:
Oh that's from CursorShapes directly from the qt documentation here https://doc.qt.io/qt-5/qt.html#CursorShape-enum. Has nothing to do with Adjustable Containers.

Reply:
Yeah... but at the end of the day we're responsible for how things are, can't point the finger at someone else. Anyway yes, nothing to do with this PR!

Reply:
Happy to add better names for CursorShapes. If someone more creative and with better language skills then me makes some suggestions :smiley:

Reply:
Sure, I've thought of them here: https://github.com/Mudlet/Mudlet/issues/3607